### PR TITLE
Add 1.6 to CI

### PR DESCRIPTION
**Pull request by odow**
A PR to track fixes for 1.6. Tests fail at present.

### CODE REVIEW

**Comment by odow** [comment]
Current failure is caused by: https://github.com/JuliaLang/julia/pull/37065. A fix is proposed: https://github.com/JuliaLang/julia/pull/38250

TODO: 
- [ ] retest when 38250 is merged

**Comment by odow** [comment]
Tests pass on Julia 1.6, but since Github actions are missing `allow-failure`, I commented out so we can merge and tag a new release. I don't think we need to test nightly permanently.

**Comment by blegat** [comment]
> I commented out so we can merge and tag a new release. I don't think we need to test nightly permanently.

The good thing is that the different CI runs are in separate checks so we can easily see if only nightly is failing.

File: test/dummy.jl [hunk 1]
```diff
@@ -9,6 +9,8 @@ DummyBigInt(J::UniformScaling) = DummyBigInt(J.λ)
 Base.ndims(::Type{DummyBigInt}) = 0
 Base.broadcastable(x::DummyBigInt) = Ref(x)
 
+Base.ndims(::DummyBigInt) = 0  # Needed in LinearAlgebra.
```
Comment (blegat) [inline test/dummy.jl:12]: In fact `ndims` is already defined above with the type, let's merge the two

**Pull request merged**